entities/src/main/java/org/odk/collect/entities/javarosa/parse/EntityFormParseProcessor.java

```diff
@@ -54,11 +56,17 @@ public void processBindAttribute(String name, String value, DataBinding binding)
 
     @Override
     public void processFormDef(FormDef formDef) throws XFormParser.ParseException {
-        if (!versionPresent && EntityFormParser.getEntityElement(formDef.getMainInstance()) != null) {
-            throw new XFormParser.MissingModelAttributeException(ENTITIES_NAMESPACE, "entities-version");
+        if (isEntityForm(formDef)) {
+            if (version == null) {
+                throw new XFormParser.MissingModelAttributeException(ENTITIES_NAMESPACE, "entities-version");
+            } else if (Stream.of(LOCAL_ENTITY_VERSIONS).anyMatch(version::startsWith)) {
+                EntityFormExtra entityFormExtra = new EntityFormExtra(saveTos);
+                formDef.getExtras().put(entityFormExtra);
+            }
         }
+    }
 
-        EntityFormExtra entityFormExtra = new EntityFormExtra(saveTos);
-        formDef.getExtras().put(entityFormExtra);
+    private static boolean isEntityForm(FormDef formDef) {
```

Comment:
There is no reason to make it static I think.

Reply:
Android Studio/IntelliJ will often auto create these methods as `static` if they can and that was the case here. I'd argue that static private members are preferable to non-static ones as they prevent object state from leaking into code that doesn't need it.